math/vc_soa/include/algebra/math/impl/vc_soa_math.hpp

```diff
@@ -0,0 +1,151 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2024 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+#pragma once
+
+// Vc include(s).
+#ifdef _MSC_VER
+#pragma warning(push, 0)
+#endif  // MSVC
+#include <Vc/Vc>
+#ifdef _MSC_VER
+#pragma warning(pop)
+#endif  // MSVC
+
+// System include(s)
+#include <algorithm>
+#include <cmath>
+
+namespace algebra::math_soa {
+
+/// Math functions on single values
+/// @{
+using std::abs;
+using std::acos;
+using std::asin;
+using std::atan;
+using std::atan2;
+using std::atanh;
+using std::ceil;
+using std::copysign;
+using std::cos;
+using std::cosh;
+using std::exp;
+using std::fabs;
+using std::floor;
+using std::fma;
+using std::hypot;
+using std::log;
+using std::log10;
+using std::max;
+using std::min;
+using std::pow;
+using std::signbit;
+using std::sin;
+using std::sinh;
+using std::sqrt;
+using std::tan;
+using std::tanh;
+/// @}
+
+/// Vc overloads of common math functions
+/// @{
+template <typename T>
+requires Vc::Traits::is_simd_vector<T>::value inline decltype(auto) abs(
```

Comment:
I'm asking this as a concept newbie: Would it not be nicer to define a concept with the Vc type trait, and use that in this declaration instead?

Reply:
Done